### PR TITLE
feat: render file tree in legacy html-based QuickLook preview

### PR DESCRIPTION
**Pull request by nevack**
Do so by constructing temporary tree-like data structure.

How to test:
```sh
qlmanage -c com.bittorrent.torrent \
         -g <path/to/built/QuickLookPlugin.qlgenerator> \
         -p <path/to/file.torrent>

```

How it looks:
image

Test file:
[temp.torrent.zip](https://github.com/transmission/transmission/files/12841894/temp.torrent.zip)

### CODE REVIEW

**Comment by nevack** [comment]
I actually do not know who to ask for review 🙃 
Maybe @tearfur has something to say about my not so good C++, and @Coeur for macOS review.

I have another PR about QuickLook in draft, but there's a lot of work to be done and I have not enough resources right now.
I thought to give some love to legacy HTML-based QuickLook preview plugin, as new QuickLook app extension won't be there soon.

As for solution I propose - I think this could be done without using Tree structure and memory overhead it introduces, but this way is just more straightforward.

I think it's possible to save prev path and check common prefix between it and new path.
This way we can tell common parent dir, and adjust `level` accordingly.
Dunno if it is worth it.

**Review by tearfur — COMMENTED** [review]
I genuinely am not familiar with the Mac client and so I have no idea why this is needed.

So I'll just stick with generic code comments.

File: macosx/QuickLookPlugin/GeneratePreviewForURL.mm [hunk 1]
```diff
@@ -13,6 +14,60 @@
...
+class FileTreeNode
+{
+  public:
+    FileTreeNode() = default;
+    ~FileTreeNode() = default;
+
+    auto MaybeCreateChild(std::string const& child_name)
+    {
+        return children_.try_emplace(child_name);
+    }
```
[the remark this answers is not in the document]
Comment (nevack) [reply macosx/QuickLookPlugin/GeneratePreviewForURL.mm]: Done!

File: macosx/QuickLookPlugin/GeneratePreviewForURL.mm [hunk 2]
```diff
@@ -13,6 +14,60 @@
 void CancelPreviewGeneration(void* thisInterface, QLPreviewRequestRef preview);
 QL_EXTERN_C_END
 
+static NSUInteger const kIconWidth = 16;
+
+namespace
+{
+
+static std::vector<std::string> SplitString(std::string_view str, std::string_view delimiter)
```
[the remark this answers is not in the document]
Comment (nevack) [reply macosx/QuickLookPlugin/GeneratePreviewForURL.mm]: Done!

**Comment by nevack** [comment]
> I genuinely am not familiar with the Mac client and so I have no idea why this is needed.

For comparison, how it looks now:

image

**Review by Coeur — APPROVED** [review]
It fixes `#warning display folders?`?
Approved. :)

**Comment by nevack** [comment]
> It fixes `#warning display folders?`?

Yep, it does. Sometimes these warnings are useful, this one in particular got on my nerves enough 😄

**Comment by nevack** [comment]
Force pushing code style issue fix

**Review by tearfur — APPROVED** [review]
C++ code LGTM.

**Comment by ckerr** [comment]
@nevack please edit the PR description at the top of this page to add a `Notes:` paragraph

**Pull request merged**